team/bundles/org.eclipse.compare/compare/org/eclipse/compare/internal/DocLineComparator.java

```diff
@@ -303,5 +305,15 @@ private boolean compare(String s1, String s2) {
 		}
 		return true;
 	}
+
+	private static boolean isSignificantWhitespace(String line, int columnNumber, char character, int lineNumber,
+			DocLineComparator comparator) {
+		boolean isWhitespace = Character.isWhitespace(character);
+		if (isWhitespace && comparator.fIgnoreWhitespaceContributor != null
```

Comment:
May I suggest to avoid `== false` comparision and instead something like 

```
		IIgnoreWhitespaceContributor ignoreWhitespaceContributor = comparator.fIgnoreWhitespaceContributor;
		if (isWhitespace && ignoreWhitespaceContributor != null
				&& !ignoreWhitespaceContributor.isIgnoredWhitespace(lineNumber, columnNumber)) {
			isWhitespace = false; // whitespace should be included and not ignored
		}
```